packages/components/chip/src/Chip.tsx

```diff
@@ -1,7 +1,105 @@
-import { ComponentPropsWithoutRef, forwardRef, PropsWithChildren } from 'react'
+import React, {
+  ButtonHTMLAttributes,
+  Children,
+  FC,
+  forwardRef,
+  isValidElement,
+  PropsWithChildren,
+  ReactElement,
+} from 'react'
 
-export type ChipProps = ComponentPropsWithoutRef<'div'>
+import { chipContentStyles, chipStyles, type ChipStylesProps } from './Chip.styles'
+import { ChipContext } from './useChipContext'
+import { useChipElement } from './useChipElement'
 
-export const Chip = forwardRef<HTMLDivElement, PropsWithChildren<ChipProps>>((props, ref) => {
-  return <div ref={ref} {...props} />
-})
+const getDisplayName = (element?: ReactElement) => {
+  return element ? (element.type as FC).displayName : ''
+}
+
+const findElement =
+  (...values: string[]) =>
+  (children: React.ReactNode) => {
+    const validChildren = Children.toArray(children).filter(isValidElement)
+
+    return validChildren.find(child => values.includes(getDisplayName(child) || ''))
+  }
```

Comment:
Taken from Input
I think that part might be moved as a util in internal-utils
What do you think @andresz1 ?

Reply:
where and how it must be named
[directory-name]/[filename].ts